core/modules/replica/HttpProcessor.h

```diff
@@ -131,119 +134,233 @@ class HttpProcessor
      * @return the context string to be used when logging messages into
      * a log stream.
      */
-    std::string context() const;
+    std::string _context() const;
 
     /**
      * Log a message into the Logger's LOG_LVL_INFO stream
      *
      * @param msg
      *   a message to be logged
      */
-    void info(std::string const& msg) {
-        LOGS(_log, LOG_LVL_INFO, context() << msg);
-    }
+    void _info(std::string const& msg);
 
     /**
      * Log a message into the Logger's LOG_LVL_DEBUG stream
      *
      * @param msg
      *   a message to be logged
      */
-    void debug(std::string const& msg) {
-        LOGS(_log, LOG_LVL_DEBUG, context() << msg);
-    }
+    void _debug(std::string const& msg);
 
     /**
      * Log a message into the Logger's LOG_LVL_ERROR stream
      *
      * @param msg
      *   a message to be logged
      */
-    void error(std::string const& msg) {
-        LOGS(_log, LOG_LVL_ERROR, context() << msg);
-    }
+    void _error(std::string const& msg);
 
-    // --------------------------------------
-    // Callbacks for processing test requests
-    // --------------------------------------
+    // --------------------------------
+    // Callback for processing requests
+    // --------------------------------
 
     /**
-     * Process "POST" requests
+     * Process a request which return status of one worker.
      *
      * @param req   request received from a client
      * @param resp  response to be sent back
      */
-    void _testCreate(qhttp::Request::Ptr req,
-                     qhttp::Response::Ptr resp);
+    void _getWorkerStatus(qhttp::Request::Ptr req,
+                          qhttp::Response::Ptr resp);
 
     /**
-     * Process "GET" requests for a list of resources of the given type
+     * Process a request which return the status of the replicas.
      *
      * @param req   request received from a client
      * @param resp  response to be sent back
      */
-    void _testList(qhttp::Request::Ptr req,
-                   qhttp::Response::Ptr resp);
+    void _getReplicationLevel(qhttp::Request::Ptr req,
```

Comment:
is "level" the same as "status", or would it be more clear name the function `_getReplicationStatus`?

Reply:
The method returns a JSON object presenting numbers of chunks at various replication levels. I tried to avoid using 'status' in this context because this term is already heavily use in related contexts, including:
* a status of a replication request
* a status of a Job
* a status of the Replication system
Though, I would be happy to rename the method into its plural form `_getReplicationLevels`. This in turn would have another counter-argument - the REST APIs are typically modeled around objects (nouns), like:
```
/service/object`
/service/object/:id`
``` 
And HTTP methods (`GET`, `PUT`,'POST` and `DELETE`) used for operations over objects. This particular method is a handler of `GET` requests to the following REST URI  `/replication/v1/level`.